lib/onix/product.rb

```diff
@@ -555,6 +555,12 @@ def supplies(keep_all_prices_dates=false)
             else
                 market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
             end
+          elsif ps.markets && !ps.available?
+            if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
+                market_territories = []
+            else
+                market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
+            end
           else
             market_territories = []
           end
```

Comment:
I am having a hard time understanding these conditions. It looks like:
```
if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty?
    market_territories = []
else
    market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
end
```
is equivalent to:
```
market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
```

Would it be easier to read as:
```
if ps.markets
    if ps.markets.map{|m| m.territory.countries}.flatten.uniq.empty? && ps.available?
        market_territories = ONIX::Territory.region_to_countries("WORLD")
    else
        market_territories = ps.markets.map{|m| m.territory.countries}.flatten.uniq
    end
else
    market_territories = []
end
```

Ie. we have a specific condition when the product is available but as no market defined. Also we should put a long comment to explain the reason for this special case.

Reply:
yes, you're right, and I have simplified the code a little more